### PR TITLE
PYTHON-3355 Test with consistent versions of crypt_shared and server

### PR DESCRIPTION
**Pull request by juliusgeo**

### CODE REVIEW

File: .evergreen/config.yml [hunk 1]
```diff
@@ -2316,6 +2317,7 @@ buildvariants:
         encryption: [ "encryption_crypt_shared" ]
       then:
         remove_tasks:
+          - ".rapid"
```
Comment (ShaneHarvey) [inline .evergreen/config.yml:2320]: We still want to test rapid versions.

File: .evergreen/run-tests.sh [hunk 2]
```diff
@@ -151,12 +152,17 @@ if [ -n "$TEST_ENCRYPTION" ]; then
     . $DRIVERS_TOOLS/.evergreen/csfle/set-temp-creds.sh
 
     if [ -n "$TEST_CRYPT_SHARED" ]; then
-        echo "Testing CSFLE with crypt_shared lib"
-        $PYTHON $DRIVERS_TOOLS/.evergreen/mongodl.py --component crypt_shared \
-            --version latest --out ../crypt_shared/
-        export DYLD_FALLBACK_LIBRARY_PATH=../crypt_shared/lib/:$DYLD_FALLBACK_LIBRARY_PATH
-        export LD_LIBRARY_PATH=../crypt_shared/lib:$LD_LIBRARY_PATH
-        export PATH=../crypt_shared/bin:$PATH
+      REAL_VERSION="$(mongod --version | head -n1 | cut -d v -f3 | cut -d - -f1 -f2)"
```
Comment (ShaneHarvey) [inline .evergreen/run-tests.sh:155]: remove `cut -d - -f1 -f2` since we need to keep the -rcX piece.

File: .evergreen/run-tests.sh [hunk 3]
```diff
@@ -151,12 +152,17 @@ if [ -n "$TEST_ENCRYPTION" ]; then
...
+      if [ "$MONGODB_VERSION" = "latest" ]; then
+        REAL_VERSION="latest"
+      fi
+      echo "Testing CSFLE with crypt_shared lib"
+      $PYTHON $DRIVERS_TOOLS/.evergreen/mongodl.py --component crypt_shared \
+          --version "$REAL_VERSION"  \
+          --out ../crypt_shared/
+      export DYLD_FALLBACK_LIBRARY_PATH=../crypt_shared/lib/:$DYLD_FALLBACK_LIBRARY_PATH
+      export LD_LIBRARY_PATH=../crypt_shared/lib:$LD_LIBRARY_PATH
+      export PATH=../crypt_shared/bin:$PATH
```
Comment (ShaneHarvey) [inline .evergreen/run-tests.sh:165]: Indent 4 spaces instead of 2.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: .evergreen/run-tests.sh (hunk 2 above)
Comment (ShaneHarvey) [reply .evergreen/run-tests.sh]: Removing it has the same behavior:
```
$ mongod --version | head -n1 | cut -d v -f3 
6.0.0-rc4
```
Let's remove it

File: .evergreen/config.yml (hunk 1 above)
Comment (ShaneHarvey) [reply .evergreen/config.yml:2320]: Oh interesting, what if we update rapid to r6.1.0-alpha? To do that you create a PR like this: https://github.com/mongodb-labs/drivers-evergreen-tools/pull/217

**Comment by blink1073** [comment]
[Error](https://evergreen.mongodb.com/task_log_raw/mongo_python_driver_test_macos_encryption__platform~macos_1100_auth~auth_ssl~nossl_encryption~encryption_crypt_shared_test_6.0_sharded_cluster_patch_3204290e93594f8ffe537ca5e7ed071d4c13056e_62f5093861837d540618fa0e_22_08_11_13_50_49/0?type=T#L1748) looks relevant: `pymongocrypt.errors.MongoCryptError: csfle lib_create() failed: Cannot initialize the Mongo Crypt Shared Support Library`

**Comment by ShaneHarvey** [comment]
Full error:
```
 [2022/08/11 14:04:59.786] ERROR: test_legacy_fle2_validatorAndPartialFieldExpression_create_with_a_validator_on_an_encrypted_field_is_an_error (test.test_encryption.TestSpec)
 [2022/08/11 14:04:59.786] ----------------------------------------------------------------------
 [2022/08/11 14:04:59.786] Traceback (most recent call last):
 [2022/08/11 14:04:59.786]   File "/System/Volumes/Data/data/mci/8f5831c6762ad7c1185896252cbde96b/src/test/__init__.py", line 596, in wrap
 [2022/08/11 14:04:59.786]     return f(*args, **kwargs)
 [2022/08/11 14:04:59.786]   File "/System/Volumes/Data/data/mci/8f5831c6762ad7c1185896252cbde96b/src/test/__init__.py", line 596, in wrap
 [2022/08/11 14:04:59.786]     return f(*args, **kwargs)
 [2022/08/11 14:04:59.786]   File "/System/Volumes/Data/data/mci/8f5831c6762ad7c1185896252cbde96b/src/test/test_encryption.py", line 664, in run_scenario
 [2022/08/11 14:04:59.786]     self.run_scenario(scenario_def, test)
 [2022/08/11 14:04:59.786]   File "/System/Volumes/Data/data/mci/8f5831c6762ad7c1185896252cbde96b/src/test/utils_spec_runner.py", line 520, in run_scenario
 [2022/08/11 14:04:59.786]     client = rs_client(
 [2022/08/11 14:04:59.786]   File "/System/Volumes/Data/data/mci/8f5831c6762ad7c1185896252cbde96b/src/test/utils.py", line 576, in rs_client
 [2022/08/11 14:04:59.786]     return _mongo_client(h, p, **kwargs)
 [2022/08/11 14:04:59.786]   File "/System/Volumes/Data/data/mci/8f5831c6762ad7c1185896252cbde96b/src/test/utils.py", line 556, in _mongo_client
 [2022/08/11 14:04:59.786]     return MongoClient(uri, port, **client_options)
 [2022/08/11 14:04:59.786]   File "/System/Volumes/Data/data/mci/8f5831c6762ad7c1185896252cbde96b/src/pymongo/mongo_client.py", line 848, in __init__
 [2022/08/11 14:04:59.786]     self._encrypter = _Encrypter(self, self.__options.auto_encryption_opts)
 [2022/08/11 14:04:59.786]   File "/System/Volumes/Data/data/mci/8f5831c6762ad7c1185896252cbde96b/src/pymongo/encryption.py", line 336, in __init__
 [2022/08/11 14:04:59.786]     self._auto_encrypter = AutoEncrypter(
 [2022/08/11 14:04:59.786]   File "/System/Volumes/Data/data/mci/8f5831c6762ad7c1185896252cbde96b/src/venv-encryption/lib/python3.9/site-packages/pymongocrypt/auto_encrypter.py", line 31, in __init__
 [2022/08/11 14:04:59.786]     self.mongocrypt = MongoCrypt(mongo_crypt_opts, callback)
 [2022/08/11 14:04:59.786]   File "/System/Volumes/Data/data/mci/8f5831c6762ad7c1185896252cbde96b/src/venv-encryption/lib/python3.9/site-packages/pymongocrypt/mongocrypt.py", line 179, in __init__
 [2022/08/11 14:04:59.786]     self.__init()
 [2022/08/11 14:04:59.786]   File "/System/Volumes/Data/data/mci/8f5831c6762ad7c1185896252cbde96b/src/venv-encryption/lib/python3.9/site-packages/pymongocrypt/mongocrypt.py", line 243, in __init
 [2022/08/11 14:04:59.786]     self.__raise_from_status()
 [2022/08/11 14:04:59.786]   File "/System/Volumes/Data/data/mci/8f5831c6762ad7c1185896252cbde96b/src/venv-encryption/lib/python3.9/site-packages/pymongocrypt/mongocrypt.py", line 259, in __raise_from_status
 [2022/08/11 14:04:59.786]     raise exc
 [2022/08/11 14:04:59.786] pymongocrypt.errors.MongoCryptError: csfle lib_create() failed: Cannot initialize the Mongo Crypt Shared Support Library when it is already initialized. [Error 3, code 1]
```

Strange, can you reproduce this error locally with 6.0.0 crypt_shared and 6.0.0 sharded cluster?

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**